### PR TITLE
Pull request for #44: Implement 4.0 test comments for readme

### PR DESCRIPTION
**Pull request by arosenbe**
@stanfordquan, can you please PR my work from #44.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,52 +1,84 @@
...
+    * [gslab_python](https://github.com/gslab-econ/gslab_python) version 4.0.0 or later of our python module with custom scons builders. 
+    * [PyYAML](http://pyyaml.org/wiki/PyYAML) a python module for parsing yaml files. 
+* [SCons](http://scons.org/pages/download.html). We recommend version 2.4 or later, but any will do.  
+* [git](https://git-scm.com/downloads) for version control.
+    * [git-lfs](https://git-lfs.github.com/) for versioning large files. 
+    * You'll need both git and git-lfs to clone the repository. 
+* [LyX](https://www.lyx.org/Download) (with instructions for LaTeX) the document processor. 
+    * Add LyX to your PATH for [windows](http://www.computerhope.com/issues/ch000549.htm), [mac](http://hathaway.cc/post/69201163472/how-to-edit-your-path-environment-variables-on-mac), and [linux](http://stackoverflow.com/questions/14637979/how-to-permanently-set-path-on-linux).
+* [Stata](http://www.stata.com/) a proprietary data analysis software.
```
Comment (stanfordquan) [inline README.md]: probably don't need to explain what Stata is

File: README.md [hunk 2]
```diff
@@ -1,52 +1,84 @@
...
+* [SCons](http://scons.org/pages/download.html). We recommend version 2.4 or later, but any will do.  
+* [git](https://git-scm.com/downloads) for version control.
+    * [git-lfs](https://git-lfs.github.com/) for versioning large files. 
+    * You'll need both git and git-lfs to clone the repository. 
+* [LyX](https://www.lyx.org/Download) (with instructions for LaTeX) the document processor. 
+    * Add LyX to your PATH for [windows](http://www.computerhope.com/issues/ch000549.htm), [mac](http://hathaway.cc/post/69201163472/how-to-edit-your-path-environment-variables-on-mac), and [linux](http://stackoverflow.com/questions/14637979/how-to-permanently-set-path-on-linux).
+* [Stata](http://www.stata.com/) a proprietary data analysis software. 
+    * Add Stata to your PATH for [windows](http://www.computerhope.com/issues/ch000549.htm), [mac](http://hathaway.cc/post/69201163472/how-to-edit-your-path-environment-variables-on-mac), and [linux](http://stackoverflow.com/questions/14637979/how-to-permanently-set-path-on-linux).
+    * [yaml](https://github.com/gslab-econ/gslab_stata) a Stata ado file for parsing yaml files. FIX THIS URL.
```
Comment (stanfordquan) [inline README.md]: FIX IT

File: README.md [hunk 3]
```diff
@@ -1,52 +1,84 @@
...
+
+## Getting started
+
+1. Open a shell, clone the repository, and navigate to its root.
+    ```
+    git clone https://github.com/gslab-econ/template.git
+    cd template
+    ```
+2. You're ready to go. We'll prompt you to enter any information necessary and store it in `user-config.yaml` as your scripts run. 
+    * To build everything that has been modified or with dependencies in the repository that have been modified.
```
Comment (stanfordquan) [inline README.md]: this is not yet a feature, correct?

File: README.md [hunk 4]
```diff
@@ -1,52 +1,84 @@
...
+
+Each user is allowed to have different local specifications: We don't put any restrictions on where you keep large files, what you call your executables, or how you manage shared directories. We do need to find these things, and that's what `user-config.yaml` is for. Each user maintains an unversioned [yaml file](http://yaml.org/) with these sorts of specifications. Each script uses its associated yaml-parsing module to read these specifications each time the script is run. 
+
+##### What do I put in `user-config.yaml`?
+
+There's no "default" for `user-config.yaml` because it depends on system specifications and user preferences. Two things we do recommend keeping in `user-config.yaml` are the name of your Stata executable—which we'll prompt you to enter if necessary—and the location of a [SCons cache directory](http://scons.org/doc/2.0.1/HTML/scons-user/c4213.html)—if you have one. An example where Example_User is running a factory-fresh StataMP and has local access to a directory named cache/template on Dropbox would be 
+```YAML
+stata_exec: statamp # StataMP-64.exe on windows
+cache: /Users/Example_User/Dropbox/cache/template
 ```
```
Comment (stanfordquan) [inline README.md]: make sure leaving the comment in doesn't break the YAML file should a real user neglect to delete it

File: README.md [hunk 5]
```diff
@@ -1,52 +1,84 @@
...
 
-#### License
+##### Can I only use Stata for data analysis?
+
```
Comment (stanfordquan) [inline README.md]: I'd suggest phrasing the question in a positive way, like "can we use other softwares" instead

File: README.md [hunk 6]
```diff
@@ -1,52 +1,84 @@
...
+Nope, we have custom builders for Python and R, and you can use them with the same syntax as the Stata builder. If you're using R, make sure it's been added to your PATH and that you have a yaml-parsing package, such as [yaml](https://cran.r-project.org/web/packages/yaml/yaml.pdf). 
+
+##### Can I write in LaTeX?
+
```
Comment (stanfordquan) [inline README.md]: write my paper

File: README.md [hunk 7]
```diff
@@ -1,52 +1,84 @@
...
-- First, either:
-	- Fork this repository
-	- Create an empty repository in GitHub and clone it locally. Copy the contents of this template into the empty repository. Make sure to exclude the `.git` folder, but include the `.gitattributes` and `.gitignore` files. Re-run the entire directory using `Scons`. Commit and push to the new repository.
-- Setup a `user-config.yaml` in the root of the directory with at least the following (note that this file should not be versioned):
-```
-stata_flavor: statamp
-cache: /Users/leviboxell/Google Drive/cache/template
+# GSLab Template
+
+The GSLab Template is a minimum working example of the tools and organization used by projects in the GSLab. We use SCons and a few custom builders to execute scripts and track dependencies in a portable and flexible manner.
```
Comment (arosenbe) [inline README.md]: How's "minimal working demonstration" to avoid repeating template?

File: README.md (hunk 1 above)
Comment (arosenbe) [inline README.md]: Thanks, also stopped explaining what LyX is.

File: README.md (hunk 2 above)
Comment (arosenbe) [inline README.md]: See [here](https://github.com/gslab-econ/stata-misc/issues/)

File: README.md (hunk 4 above)
Comment (arosenbe) [inline README.md]: PyYAML skips comments when parsing YAML file. The parser in stata does not.

File: README.md (hunk 6 above)
Comment (arosenbe) [inline README.md]: The answer's still yes.

File: README.md (hunk 3 above)
Comment (arosenbe) [inline README.md]: This will be soon. Waiting on #45.

**Comment by arosenbe** [comment]
On hold

**Comment by arosenbe** [comment]
On hold

File: README.md (hunk 3 above)
Comment (arosenbe) [inline README.md]: @stanfordquan, We've pulled in #45 so I think this is ready to go.

**Comment by arosenbe** [comment]
@stanfordquan, can you take another look at this? I caught a few typos/poor grammar on my last read through. It might make sense for you to take a pretty detailed look.

**Comment by stanfordquan** [comment]
Looks good, thanks @arosenbe. The merge conflict looks simple, but I'll take another look after you have pulled it in. Thanks!

**Comment by arosenbe** [comment]
@stanfordquan, can you look again. I think the only issue was in the yaml example in the readme, but I'd like independent confirmation.

**Comment by stanfordquan** [comment]
I removed a conflict flag and squash-merged.

**Pull request merged**